src/main/java/edu/kit/provideq/toolbox/maxcut/solvers/QiskitMaxCutSolver.java

```diff
@@ -53,12 +58,45 @@ public void solve(Problem<String> problem, Solution<String> solution,
         .addSolutionFilePathToProcessCommand()
         .run(problem.type(), solution.getId(), problem.problemData());
 
-    if (processResult.success()) {
-      solution.setSolutionData(processResult.output());
-      solution.complete();
-    } else {
+    // Return if process failed
+    if (!processResult.success()) {
       solution.setDebugData(processResult.output());
       solution.abort();
+      return;
     }
+
+    // Parse GML to add partition data to
+    Gml gml;
+    try {
+      gml = Gml.fromString(problem.problemData());
+    } catch (ConversionException e) {
+      solution.setDebugData("Couldn't convert problem data to GML:\n" + e);
+      solution.abort();
+      return;
+    }
+
+    // Parse solution data and add partition data to GML
+    Optional<String> solutionLine = processResult.output()
+            .lines()
+            .filter(s -> s.startsWith(SOLUTION_LINE_PREFIX))
+            .findFirst();
+
+    if (solutionLine.isPresent()) {
+      // Prepare solution data from python output
+      String s = solutionLine.get();
+      var solutionData = s
+              .substring(SOLUTION_LINE_PREFIX.length() + 1, s.length() - 1)
```

Comment:
the `endIndex` in `subString` is exclusive, shouldn't it be `s.length()` instead?

Edit: I see, you're removing the brackets here. Can you add a comment explaining these +/-1s?